Objects/bytearrayobject.c

```diff
@@ -899,6 +896,13 @@ bytearray___init___impl(PyByteArrayObject *self, PyObject *arg,
     PyObject *it;
     PyObject *(*iternext)(PyObject *);
 
+    /* First __init__; set ob_bytes_object so ob_bytes is always non-null. */
```

Comment:
`__init__` can be called several times from Python code, but also *skipped*:
```
>>> ba = bytearray.__new__(bytearray)
>>> ba.append(123)
```
Invariants like this need to be introduced in `tp_init`.